### PR TITLE
Migrate: add package dependencies to test projects

### PR DESCRIPTION
**Pull request by jgoshi**
Adding references to Microsoft.Net.Test.Sdk, xunit, xunit.runner.visualstudio

### CODE REVIEW

**Comment by jgoshi** [comment]
@piotrpMSFT @livarcocc @krwq

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs [hunk 1]
```diff
@@ -99,6 +99,31 @@ public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migra
...
+
+                    _transformApplicator.Execute(
+                        PackageDependencyInfoTransform.Transform(
+                            new PackageDependencyInfo
+                            {
+                                Name = PackageConstants.XUnitPackageName,
+                                Version = PackageConstants.XUnitPackageVersion
+                            }), noFrameworkPackageReferenceItemGroup, mergeExisting: false);
+
+                    _transformApplicator.Execute(
```
Comment (livarcocc) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs]: I don't think you can blindly add the xunit package. You should only do that if the test project already had xunit in it.

File: test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackageDependencies.cs [hunk 2]
```diff
@@ -258,6 +258,28 @@ public void It_auto_add_desktop_references_during_migrate()
             items[0].Include.Should().Be("System");
         }
 
+        [Fact]
+        public void It_migrates_test_projects_to_have_test_sdk_and_xunit_packagedependencies()
```
Comment (livarcocc) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackageDependencies.cs]: This should be two separate tests:

One to add the test sdk
One to convert the xunit dependency to the new packages.

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs (hunk 1 above)
Comment (jgoshi) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs]: I added two unit tests, are those good enough?

File: src/Microsoft.DotNet.ProjectJsonMigration/ProjectType.cs [hunk 3]
```diff
@@ -3,6 +3,7 @@ namespace Microsoft.DotNet.ProjectJsonMigration
     internal enum ProjectType
     {
         Console,
-        Web
+        Web,
+        Test
```
Comment (jgoshi) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectType.cs]: This handles a test projects. I only inject the xunit related dependencies if the testrunner is xunit.

**Pull request merged**